### PR TITLE
Update the default kv factory to kv.Factory

### PR DESCRIPTION
**Pull request by VioletHynes**
This is the result of a discovery that the default `secret` mount in tests weren't sending any events, and the subsequent discovery that while the default `secret` mount was KVv1 for non-tests, it was a `LeasedPassthroughBackend` (similar, but not quite) for tests. 

Updating the logical backend "kv" to `kv.Factory` fixes this disparity, but this raised the question as to why `LeasedPassthroughBackendFactory` is the default. As far as I can tell, this won't change any non-test code behaviour, and I can't find a way to make a `LeasedPassthroughBackend` in a dev/non-test cluster. That being said, there's obviously a lot of history here, so I want to ensure that assumption is challenged.

The salient change is the single-line code change in `core.go`.

The default `secret` mount is already KV, so there's no change there, and it's another case of this change ensuring that our tests are closer to 'reality':
`secret/       kv           kv_42215ab2           key/value secret storage`

There were some tests that depended on this functionality, so I set the `kv` factory to be `LeasedPassthroughBackendFactory` for those tests. This makes the default `kv` to be `kv.Factory`, which is more intuitive and also tests closer to what a customer's Vault instance is like. This technical debt bit me when I was working on static secret caching for Proxy, and will likely bite future developers, too.

### CODE REVIEW

File: vault/testing.go [hunk 1]
```diff
@@ -2143,28 +2152,7 @@ func (tc *TestCluster) initCores(t testing.T, opts *TestClusterOptions, addAudit
...
-		},
-	}
-	resp, err := leader.Core.HandleRequest(namespace.RootContext(ctx), kvReq)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if resp.IsError() {
-		t.Fatal(err)
-	}
+	testCoreAddSecretMount(t, leader.Core, tc.RootToken, kvVersion)
```
Comment (VioletHynes) [inline vault/testing.go:2155]: This refactor technically isn't related to the default kv factory, but it was hard not to notice while making/investigating the change, and seemed like an easy win.

**Review by swenson — APPROVED** [review]
LGTM

I definitely appreciate minimizing the differences between all the different ways that cores can be configured during tests.

**Comment by ncabatoff** [comment]
Can you try making an ENT PR for this change, to see if there are any other tests that will break?

**Comment by VioletHynes** [comment]
> Can you try making an ENT PR for this change, to see if there are any other tests that will break?

Good call, I will do!

**Review by ncabatoff — APPROVED** [review]

**Pull request merged**